Sources/Swift/Integrations/SessionReplay/SentryTouchTracker.swift

```diff
@@ -34,39 +35,48 @@ class SentryTouchTracker: NSObject {
      * will ever have the same pointer.
      */
     private var trackedTouches = [UITouch: TouchInfo]()
+    private let dispatchQueue: SentryDispatchQueueWrapper
     private var touchId = 1
     private let dateProvider: SentryCurrentDateProvider
     private let scale: CGAffineTransform
     
-    init(dateProvider: SentryCurrentDateProvider, scale: Float) {
+    init(dateProvider: SentryCurrentDateProvider, scale: Float, dispatchQueue: SentryDispatchQueueWrapper) {
         self.dateProvider = dateProvider
         self.scale = CGAffineTransform(scaleX: CGFloat(scale), y: CGFloat(scale))
+        self.dispatchQueue = dispatchQueue
+    }
+    
+    convenience init(dateProvider: SentryCurrentDateProvider, scale: Float) {
+        self.init(dateProvider: dateProvider, scale: scale, dispatchQueue: SentryDispatchQueueWrapper())
     }
```

Comment:
`m`: https://github.com/getsentry/sentry-cocoa/blob/da7a50e80c96a369ab8e53bfa59ea4efb0b57f90/Sources/Sentry/SentrySessionReplayIntegration.m#L90-L95 now uses the convenience init which creates an extra `SentryDispatchQueueWrapper`. Does this class need it's own instance? If not, I would recommend deleting the convenience init and using the `SentryDispatchQueueWrapper` from the DependencyContainer.

Reply:
SentryDispatchQueueWrapper default dispatchQueue uses a serial queue shared with every other code that uses it. I really dont think this is the way to go only to save a few bytes in memory, specially in an operation that will use its dispatch queue super often.

Reply:
> I really don’t think this is the way to go only to save a few bytes in memory, especially in an operation that will use its dispatch queue super often.

I prefer saving a few bytes in memory when it's easy possible when we don't need the extra queue. Do you think we need an extra one and if so why?

Reply:
Yes, to avoid adding another bottleneck to session replay, which is already slow.

Reply:
> Yes, to avoid adding another bottleneck to session replay, which is already slow.

Please add this info as a comment for the convenience init.

> What is the problem with SentryTouchTracker having its own dispatch queue?

A few bytes in memory, but if you want to have your own dispatch queue to ensure other stuff doesn't slow down SR, I'm fine with the tradeoff.